### PR TITLE
[AO migration] minor physics changes

### PR DESCRIPTION
**Pull request by aclegg3**
## Motivation and Context

ArticulatedObject migration PR split from #1226.
Contains all minor, straightforward changes from #1247:
- refactoring object activation API
- refactoring collision compound construction from mesh hierarchy
- doc changes

## How Has This Been Tested

Current tests should cover most of this.

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/esp/physics/PhysicsManager.h [hunk 1]
```diff
@@ -1025,41 +1025,59 @@ class PhysicsManager : public std::enable_shared_from_this<PhysicsManager> {
     return false;
   };
 
+  /**
+   * @brief Perform discrete collision detection for the scene with the derived
+   * PhysicsManager implementation. Not implemented for default @ref
+   * PhysicsManager. See @ref bullet::BulletPhysicsManager.
+   */
+  virtual void performDiscreteCollisionDetection() {
```
Comment (Skylion007) [inline src/esp/physics/PhysicsManager.h:1033]: Should this be a pure virtual? = 0?

Comment (aclegg3) [reply src/esp/physics/PhysicsManager.h:1033]: We instance the base PhysicsManager in the absence of Bullet, so it cannot be and abstract class.

File: src/esp/physics/PhysicsManager.h [hunk 2]
```diff
@@ -96,7 +96,7 @@ class PhysicsManager : public std::enable_shared_from_this<PhysicsManager> {
   PhysicsManager. Each entry suggests a derived class of @ref PhysicsManager and
   @ref RigidObject implementing the specific interface to a simulation library.
   */
-  enum PhysicsSimulationLibrary {
+  enum class PhysicsSimulationLibrary {
```
Comment (jturner65) [inline src/esp/physics/PhysicsManager.h:99]: Would now be a good time to refactor these enumerators to be Pascal case?

Comment (jturner65) [inline src/esp/physics/PhysicsManager.h:99]: Would now be a good time to refactor these enumerators to be Pascal case?

**Review by jturner65 — APPROVED** [review]
LGTM.  Just one comment regarding the enumerators for PhysicsSimulationLibrary.

File: src/esp/physics/PhysicsManager.h (hunk 2 above)
Comment (aclegg3) [reply src/esp/physics/PhysicsManager.h:99]: Fun migration. Need to work around the `None` keyword in python. Refactoring to `NoPhysics` instead.

Comment (aclegg3) [reply src/esp/physics/PhysicsManager.h:99]: BTW, I'm leaving the config and metadata tag "none" for now.

**Pull request merged**